conans/test/unittests/model/version/test_version_comparison.py

```diff
@@ -53,6 +69,19 @@ def test_comparison(v1, v2):
     assert v2 >= v1
 
 
+def test_build_metadata_is_not_equal():
+    # https://github.com/conan-io/conan/issues/5900
+    assert Version("4.0.0+abc") != Version("4.0.0+xyz")
+    # Shouldn't be an "official" order for build metadata, but as they cannot be equal
+    # the order is alphabetic
+    assert Version("4.0.0+abc") != Version("4.0.0+xyz")
```

Comment:
This assert identical to the one in the previous line? Better remove it (I know it was already there in previous file, but taking the chance and cleaning it?)